geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+This can be validated in the CRD schema.
+
+A child resource cannot `attachTo` any Gateway resource that doesn't allow attachment (eg. it does not specify `spec.infrastructure.allowedChildren` for `Gateway`s).
+This will be detected by the implementation and reported in [status](#status-fields).
+
+##### Listeners
+
+Implementations MUST treat the "parent" Gateway as having the concatenated list of all listeners from itself and "child" Gateways.
+
+Validation of this list of listeners MUST behave the same as if the list were part of a single "parent" Gateway.
+
+eg.
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  infrastructure:
+    allowedChildren:
+      namespaces:
+        from: Same
+  gatewayClassName: example
+  listeners:
+  - name: HTTP
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: child-gateway
+spec:
+  gatewayClassName: example
+  infrastructure:
+    attachTo:
+      name: parent-gateway
+      kind: Gateway
+      group: gateway.networking.k8s.io
+  listeners:
+  - name: metrics
+    port: 8080
+    protocol: HTTP
+```
+
+With this configuration, the realized "parent" Gateway should listen on port `80` and `8080`.
+
+###### Listener Precedence
+
+Gateway Listeners should be merged using the following precedence:
+- "parent" Gateway
+- "child" Gateway ordered by creation time (oldest first)
+- "child" Gateway ordered alphabetical by “{namespace}/{name}”.
+
+If there are conflicts between these, this should be reported as `Conflicted=True` in the listener as usual.
+
+### Status Fields
+
+#### Addresses
+
+The list of `Addresses` that appear in the status of the "child" Gateway MUST be the same as the "parent" Gateway.
+
+#### AttachedGateways
+
+The property `attachedGateways` will appear on the status of the "parent" Gateway will indicate the total number of "child" Gateways that have are attached
+
+#### Gateway Conditions
+
+Gateway conditions currently supports the following condition types: `Accepted` and `Programmed`
+
+For parent gateways, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a child listener is invalid, `ListenersNotValid` would be reported.
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
+
+For child gateways, `Accepted` and `Programmed` should not consider the overall merged Gateway status, but only the child's own listeners.
+The exception to this is if the parent Gateway is not valid in any regard.
+
+For example, if I have a `parent`, `child-1`, and `child-2`:
+* If `parent` is entirely invalid (for example, an invalid `address`), all three Gateways will reported `Accepted=False`.
```

Comment:
I'm not in favor of marking all three Gateways as Accepted=False if the parent is entirely invalid.  Why not let the other Gateways act as backups?

Reply:
Gateway merging is logical. The parent gateway is what creates the underlying infrastructure and the other child gateways are just a sharded list of listeners.

If a parent gateway isn't valid then there's no way to fallback onto a child listener.

Reply:
The details here may vary between implementations, but in general I view this pattern as parent creates/reserves the underlying infrastructure, and child gateways program listeners onto it.

If the infrastructure fails to be created, the behavior of accepting child gateways would at best be unpredictable (something like falling back to spinning up their own in-cluster proxy pods instead of attaching to a centralized resource, potentially cluster-external) and could easily run into issues with L3 policy rules, IP allocation, etc such that I think rejecting children of an invalid parent is likely the safer and more consistent option.